File: app/src/main/java/com/waveneuro/data/preference/PreferenceManager.java
package com.waveneuro.data.preference;

import com.waveneuro.domain.model.user.UserInfo;

public interface PreferenceManager {

    String getAccessToken();

    void setAccessToken(String accessToken);

    String getRefreshToken();

    void setRefreshToken(String refreshToken);

    void logout();

    void saveUser(UserInfo user);

    void saveTreatmentLength(String treatmentLength);

    void saveProtocolFrequency(String protocolFrequency);

    void saveProtocolId(String protocolId);

    void saveSonalId(String sonalId);

    String getTreatmentLength();

    String getProtocolFrequency();

    String getProtocolId();

    String getSonalId();

    String getUserId();

    String getFirstName();

    String getLastName();

    String getEmail();

    String getUsername();

    String getEegId();

    void saveEegId(String eegId);

    Long getPatientId();

    void savePatientId(Long eegId);

    void rememberUsername(String username);

    void rememberPassword(String password);

    String getRememberUsername();

    String getRememberPassword();

    void removeRememberUser();

    void removeRememberPassword();

    boolean getOnboardingDisplayed();

    void setOnboardingDisplayed();

    boolean getPrecautionsDisplayed();

    void setPrecautionsDisplayed();
}

File: app/src/main/java/com/waveneuro/data/preference/PreferenceManager.kt
package com.waveneuro.data.preference

import com.waveneuro.domain.model.user.UserInfo

interface PreferenceManager {
    var accessToken: String?
    var refreshToken: String?
    fun logout()
    fun saveUser(user: UserInfo)
    val user: UserInfo
    var treatmentLength: String?
    var protocolFrequency: String?
    var protocolId: String?
    var sonalId: String?
    var userId: String?
    var firstName: String?
    var lastName: String?
    var email: String?
    var username: String?
    var eegId: String?
    var patientId: Long
    var rememberUsername: String?
    var rememberPassword: String?
    fun removeRememberUser()
    fun removeRememberPassword()
    var isOnboardingDisplayed: Boolean
    var isPrecautionsDisplayed: Boolean
}
